Library/Homebrew/formula.rb

```diff
@@ -405,11 +413,19 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |pn|
+      pn.basename.to_s.start_with?("HEAD")
```

Comment:
The revision will be reset to zero once there is a stable version bump.

My previous thought is that if there is a revision bump during same stable version, we may have multi HEAD installations with the same source_modified_time. Therefore, we need to check revision number. Considering we will have check `head_prefix_up_to_date?`, I think we should be fine even the revision is reset to zero after stable version bump.

To make this method more readable, how about:
```ruby
def latest_head_prefix
  head_vesrions = {}
  installed_prefxies.each do |pn|
    pn_pkgversion = PkgVersion.parse(pn.basename)
    next unless pn_pkgversion.head?
    head_versions[pn_pkgversion] = Tab.for_keg(pn).source_modified_time
  end

  latest_source_modified_time = head_versions.values.max
  latest_head_version = head_versions.select do |v, t|
    v.revision == revision && t == latest_source_modfied_time 
  end.first
  prefix(latest_head_version) if latest_head_version
end
```